### ISSUE TITLE
Validate 2.0.7 does not detect more than one collection product in the same directory

### ISSUE THREAD

**Issue by sslavney** [NASA-PDS-Incubator/validate#405]
## 🐛 Describe the bug
When running Validate 2.0.7 on a bundle that includes three collections in the same directory, Validate finds one of the collection products but warns that it can't find the other two. It goes on to validate the products in those collections, but for each one it warns that the product does not belong to any collection. 

## 📜 To Reproduce
The bundle in question is the InSight SEIS bundle at https://pds-geosciences.wustl.edu/insight/urn-nasa-pds-insight_seis/. The data directory contains three collection products, data_lander, data_seed, and data_table. Unless Validate can operate remotely, I don't know how to give you access to the bundle to reproduce the error. The bundle is too large to send a copy. I am attaching a zip file with two Validate reports, one from version 1.24.0 without warnings and one from version 2.0.7 with warnings. Nothing changed in the bundle between the two reports. See the warning in the 2.0.7 report starting at line 360241.

## 🕵️ Expected behavior
I expected Validate to find all three collection products in the data directory and to recognize the products belonging to them.

## 📚 Version of Software Used
Validate 2.0.7

## 🩺 Test Data / Additional context
[validate.zip](https://github.com/NASA-PDS/validate/files/7187817/validate.zip)

##  🏞Screenshots
n/a

## 🖥 System Info
 - OS: Windows 10

---

**Comment by qchaupds** [comment]
@sslavney @jordanpadams 

Using the current code as of today, and running validate against test resource from github230 which has 4 distinct collection, all the collections are acknowledged:

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 153 % validate -R pds4.bundle --skip-context-validation -r report_github401_valid.json -s json -t src/test/resources/github230/valid/cocirs_c2h4abund/bundle_cocirs_c2h4abund.xml 
```
The report shows all collections are inspected:

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 151 % egrep "label" report_github401_valid.json
    "allowUnlabeledFiles": "false",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/bundle_cocirs_c2h4abund.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/context/**collection_context_cocirs_c2h4abund.xml**",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/data/cocirs_c2h4abund_abund_profiles.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/data/cocirs_c2h4abund_temp_profiles.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/data/**collection_cocirs_c2h4abund.xml**",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/document/cocirs_c2h4abund_document.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/document/cocirs_c2h4abund_document2.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/document/**collection_document_cocirs_c2h4abund.xml",**
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/xml_schema/**collection_schema_cocirs_c2h4abund.xml"**,
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/context/**collection_context_cocirs_c2h4abund.xml**",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/data/c**ollection_cocirs_c2h4abund.xml**",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/bundle_cocirs_c2h4abund.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/document/**collection_document_cocirs_c2h4abund.xml**",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/xml_schema/**collection_schema_cocirs_c2h4abund.xml**",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/data/cocirs_c2h4abund_abund_profiles.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/data/cocirs_c2h4abund_temp_profiles.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/document/cocirs_c2h4abund_document.xml",
      "label": "file:/home/qchau/sandbox/validate/src/test/resources/github230/valid/cocirs_c2h4abund/document/cocirs_c2h4abund_document2.xml",
```

Is there a way for you to try out a newer version of the code or the current snapshot of the repo to see if it fixes the issue you'd found?

https://github.com/NASA-PDS/validate/releases

These are the collection files for github230:

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 156 % find src/test/resources/github230 -name "*collection*.xml" | grep valid | grep -v invalid
src/test/resources/github230/valid/cocirs_c2h4abund/context/collection_context_cocirs_c2h4abund.xml
src/test/resources/github230/valid/cocirs_c2h4abund/data/collection_cocirs_c2h4abund.xml
src/test/resources/github230/valid/cocirs_c2h4abund/document/collection_document_cocirs_c2h4abund.xml
src/test/resources/github230/valid/cocirs_c2h4abund/xml_schema/collection_schema_cocirs_c2h4abund.xml
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 157 % 
```

**Comment by jordanpadams** [comment]
@sslavney can you try this out with the latest SNAPSHOT?  We also plan on tagging a new release in the next few days if you want to wait for that.

https://github.com/NASA-PDS/validate/releases/tag/v2.1.0-SNAPSHOT

**Comment by sslavney** [comment]
We'll give it a try in the next day or two.

**Comment by sslavney** [comment]
Looks like this particular problem has been fixed with v2.1.0-SNAPSHOT. It now sees all three collections. A different problem is turning up, though. It sees all three collections, but wants to skip one of them. That will be a separate ticket.

**Comment by jordanpadams** [comment]
copy @sslavney interested to see the issue you are seeing. our logging now outputs every collection it comes across, even if it is an older version / not referenced by the bundle. if this is still unexpected, please feel free to open a new issue.
